### ISSUE TITLE
False positive in `ReferenceEquality` for `java.lang.Thread`

### ISSUE THREAD

**Issue by PBoddington** [google/error-prone#5900]
Since bumping from version `2.49.0` to `2.50.0` we are finding that `ReferenceEquality` is now flagging the use of `==` to compare instances of `java.lang.Thread` as problematic.

For example, the class

```java
public final class MyClass {

  private final Thread threadOnWhichObjectWasCreated = Thread.currentThread();

  public boolean isSameThread() {
    return Thread.currentThread() == threadOnWhichObjectWasCreated;
  }
}
```

results in 
```
[ReferenceEquality] Comparison using reference equality instead of value equality
      return Thread.currentThread() == threadOnWhichObjectWasCreated;
                                    ^
```

I think that comparing instances of `Thread` using `==` is idiomatic Java, and that treating `Thread` as a special case might make sense. Even though `Thread` is not final, and it is technically possible to override `equals` in a subclass, it only really makes sense to treat `Thread` as an identity-based class.

We are using errorprone version `2.50.0`.

**Comment by cpovirk** [comment]
Thanks, I hit this one myself [in Guava](https://github.com/google/guava/blob/3de1f25e258ef6fd887595cc865efe185b373aa6/guava/src/com/google/common/util/concurrent/ExecutionSequencer.java#L323), and, as in your example, it was important to use `==` even if some `Thread` subclass were to override `equals`. Plus, while I haven't done an exhaustive search, my initial searches have not turned up any overrides of `Thread.equals`, anyway. I will propose a change for this.

(I suspect that ReferenceEquality will remain annoying in some cases, and we are likely to be more reluctant to make exceptions in other cases, but please keep the feedback coming.)
